File: app/src/main/java/com/lumiyaviewer/lumiya/slproto/messages/SimulatorMapUpdate.java
package com.lumiyaviewer.lumiya.slproto.messages;
import java.util.*;

import com.lumiyaviewer.lumiya.slproto.SLMessage;
import java.nio.ByteBuffer;

public class SimulatorMapUpdate extends SLMessage {
    public MapData MapData_Field = new MapData();

    public static class MapData {
        public int Flags;
    }

    public SimulatorMapUpdate() {
        this.zeroCoded = false;
    }

    public int CalcPayloadSize() {
        return 8;
    }

    public void Handle(SLMessageHandler sLMessageHandler) {
        sLMessageHandler.HandleSimulatorMapUpdate(this);
    }

    public void PackPayload(ByteBuffer byteBuffer) {
        byteBuffer.putShort(-1);
        byteBuffer.put((byte) 0);
        byteBuffer.put((byte) 5);
        packInt(byteBuffer, this.MapData_Field.Flags);
    }

    public void UnpackPayload(ByteBuffer byteBuffer) {
        this.MapData_Field.Flags = unpackInt(byteBuffer);
    }
}

File: lumiya/src/main/java/com/lumiyaviewer/lumiya/slproto/messages/SimulatorMapUpdate.kt
package com.lumiyaviewer.lumiya.slproto.messages
import java.util.*

import com.lumiyaviewer.lumiya.slproto.SLMessage
import java.nio.ByteBuffer

class SimulatorMapUpdate : SLMessage {
    MapData MapData_Field = MapData()

    class MapData {
        Int Flags
    }

    SimulatorMapUpdate() {
        this.zeroCoded = false
    }

    Int CalcPayloadSize() {
        return 8
    }

    Unit Handle(SLMessageHandler sLMessageHandler) {
        sLMessageHandler.HandleSimulatorMapUpdate(this)
    }

    Unit PackPayload(ByteBuffer byteBuffer) {
        byteBuffer.putShort(-1)
        byteBuffer.put((Byte) 0)
        byteBuffer.put((Byte) 5)
        packInt(byteBuffer, this.MapData_Field.Flags)
    }

    Unit UnpackPayload(ByteBuffer byteBuffer) {
        this.MapData_Field.Flags = unpackInt(byteBuffer)
    }
}
